glossary.ftd

```diff
@@ -1,22 +1,219 @@
+
+
+-- ft.h2: clone
+
+Clone a [server project](glossary/#server-project) into a new directory
+which becomes a [client](glossary/#server-project).
+
+
+-- ft.h2: conflict-state
+
+The file is said to be in the `conflict-state` if the file gets in conflict with
+server latest [version](glossary/#version) of the same file.
+
+```
+The file, in conflict-state, is never considered to be synced unless its
+conflict is resolved.
+```
+
+
+-- ft.h2: document
+
+The [file](glossary/#file) with extension that are compiled using FPM.
+Currently, there are two such extensions `.md` and `.ftd`.
+
+
+-- ft.h2: document-name
+
+The unique identifier of a [document](glossary/#document).
+
+This can accessed using [`document-filename`](processors/document-filename/)
+processor.
+
+
+-- ft.h2: document-full-id
+
+The `document-full-id` is same as the [`document-id`](glossary/#document-id).
+Though, in some case, it could be alias to `document-id` with a
+[`document-suffix`](glossary/#document-suffix) added after special character
+`/-/`. In case, when `document-suffix` is not present, `document-full-id` and
+`document-id` are same.
+
+For a document `foo.ftd`, the `document-id` is `/foo/` but `document-full-id`
+could be `/foo/-/x/` where `/x/` is the `document-suffix`. Both `/foo/` and
+`/foo/-/x/` points to `foo.ftd`, probably with different
+[`document-metadata`](glossary/#document-metadata).
+
+This can accessed using [`document-full-id`](processors/document-full-id/)
+processor.
+
+
+-- ft.h2: document-id
+
+The `document-id` is the url read-only property of the document interface
+returns the location from where the compiled version of document can be
+accessed. Though it can differ from [`document-full-id`](glossary/#document-full-id)
+in the respect that KVP.
+
+For a document `foo.ftd`, the `document-id` is `/foo/`
+
+This can accessed using [`document-id`](processors/document-id/)
+processor.
+
+
+-- ft.h2: document-metadata
+
+The `document-metadata` is the key-value pair data provided to
+[`document-full-id`](glossary/#document-full-id) in the
+[sitemap](glossary/#sitemap). This data can be accessed by the document, it's
+pointing to via `document-id`, using [`get-data`](processors/get-data/)
+processor.
+
+
+-- ft.code: `document-metadata` in sitemap
+lang: ftd
+
+# Foo title: /foo/
+  name: Arpita
+
+
+-- ft.markdown:
+
+Here, `/foo/` is both `document-full-id` and [`document-id`](glossary/#document-id).
+`name: Arpita` is `document-metadata` where `name` is key and `Arpita` is value.
+`foo.ftd` is the [`document-filename`](glossary/#document-filename). This
+document, `foo.ftd`, can access the `document-metadata` using `get-data`
+processor.
+
+
+
+-- ft.h2: document-suffix
+
+The `document-suffix` is the special read-only property in
+[`document-full-id`](glossary/#document-full-id) which is added after special
+character `/-/`.
+
+It helps to provide different [`document-metadata`](glossary/#document-metadata)
+to render different content.
```

Comment:
This is one reason, but not the only one. The main reason for suffix is to republish the same document in different places in sitemap. Ideally why something exists is not very relevant for glossary. Glossary should only specify what that thing is.